// @vkontakte/vkui/dist/es6/components/Button/Button.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from 'react';
import getClassName from '../../helpers/getClassName';
import classNames from '../../lib/classNames';
import Tappable from '../Tappable/Tappable';
import usePlatform from '../../hooks/usePlatform';

var Button = function Button(props) {
  var _classNames;

  var platform = usePlatform();

  var className = props.className,
      size = props.size,
      mode = props.mode,
      stretched = props.stretched,
      align = props.align,
      children = props.children,
      before = props.before,
      after = props.after,
      getRootRef = props.getRootRef,
      Component = props.Component,
      restProps = _objectWithoutProperties(props, ["className", "size", "mode", "stretched", "align", "children", "before", "after", "getRootRef", "Component"]);

  return React.createElement(Tappable, _extends({}, restProps, {
    className: classNames(getClassName('Button', platform), className, (_classNames = {}, _defineProperty(_classNames, "Button--sz-".concat(size), true), _defineProperty(_classNames, "Button--lvl-".concat(mode), true), _defineProperty(_classNames, "Button--aln-".concat(align || 'center'), true), _defineProperty(_classNames, 'Button--str', stretched), _classNames)),
    getRootRef: getRootRef,
    Component: restProps.href ? 'a' : Component
  }), React.createElement("div", {
    className: "Button__in"
  }, before && React.createElement("div", {
    className: "Button__before"
  }, before), children && React.createElement("div", {
    className: "Button__content"
  }, children), after && React.createElement("div", {
    className: "Button__after"
  }, after)));
};

Button.defaultProps = {
  mode: 'primary',
  Component: 'button',
  size: 'm',
  stretched: false,
  stopPropagation: true
};
export default Button;
